apps/web/components/dialog/EditLocationDialog.tsx

```diff
@@ -215,6 +232,147 @@ export const EditLocationDialog = (props: ISetLocationDialog) => {
           )}
         </div>
       </div>
+    ) : selectedLocation === LocationType.Whereby ? (
```

Comment:
I think all of this can be done with a lot more DRY code, cc @zomars

Reply:
Agreed. I know this is unrelated to this PR but this PR is surfacing the unmaintainability of this approach. So my take is to fix this in this PR once and for all. This can be handled with a dynamic component.